### PR TITLE
feat: Gamify applications

### PR DESCRIPTION
**Pull request by sanderelderman**
Note: Merge #240 before this one

Create all basic CRUD endpoints for the management of Applications within gamify.

### CODE REVIEW

**Comment by mschilling** [comment]
Tip: you could either keep this PR in "Draft mode" until the other branch have been merged, or have this PR target the feature branch of the other PR. This way, you can keep the changes in this PR small for all reviewers.

File: apps/gamify/api/src/main.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ async function bootstrap() {
   const app = await NestFactory.create(AppModule);
   const globalPrefix = 'api';
   app.setGlobalPrefix(globalPrefix);
+  app.useGlobalPipes(new ValidationPipe());
```
Comment (mschilling) [inline apps/gamify/api/src/main.ts:15]: 👍🏽

File: libs/gamify/applications/src/lib/dto/update-application.input.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { OmitType } from "@nestjs/mapped-types";
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/dto/update-application.input.ts:1]: Fancy

File: libs/gamify/applications/src/lib/dto/create-application.input.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @IsOptional()
+    @IsUrl()
+    externalApplicationUrl: string;
+
+    // Admin user id should not be provided by the client as input value but 
+    // should be obtained from current authenticated user.
+    @IsOptional()
+    @IsNumber()
+    adminUserId: number;
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/dto/create-application.input.ts:23]: I'm not in favour of having a property named `adminUserId`. How is this different than just `userId`?

File: libs/gamify/data/prisma/seed.ts [hunk 4]
```diff
@@ -28,10 +39,15 @@ async function seedApplications() {
...
+function getUsers() {
+  return [
+    { email: 'test@test.com' },
+  ]
+}
+
 function getApplications() {
   return [
-    { name: 'Walking Challenge' },
+    { name: 'Walking Challenge', applicationType: "CHALLANGE", adminUserId: 1 },
```
Comment (mschilling) [inline libs/gamify/data/prisma/seed.ts:50]: CHALLANGE -> CHALLENGE

File: libs/gamify/applications/src/lib/applications.controller.ts [hunk 5]
```diff
@@ -1,12 +1,76 @@
...
-    return this.applicationsService.getAll();
+  findAll() {
+    return this.applicationsService.findMany();
+  }
+
+  @Get(':id')
+  async findOne(@Param('id', ParseIntPipe) id: number, @Res() response: Response) {
+    const app = await this.applicationsService.findOne(id);
+
+    if (app === null) {
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/applications.controller.ts:37]: Can you use (built in) exception filters here?
* https://docs.nestjs.com/exception-filters

File: libs/gamify/applications/src/lib/dto/create-application.input.ts (hunk 3 above)
Comment (sanderelderman) [reply libs/gamify/applications/src/lib/dto/create-application.input.ts]: Changed it to ownerUserId for clarification

File: libs/gamify/applications/src/lib/applications.controller.ts [hunk 6]
```diff
@@ -1,28 +1,24 @@
...
+    console.log(JSON.stringify(createApplicationInput));
+
     if (! await this.applicationsService.isNameUnique(createApplicationInput.name)) {
-      response.status(HttpStatus.BAD_REQUEST).send({ 
-        'error': 'Bad Request',
-        'statusCode': 400,
-        'message': ["Name must be unique"]
-      });
-      return;
+      throw new BadRequestException("Name must be unique");
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/applications.controller.ts:15]: Nice 👌🏽

**Review by mschilling — APPROVED** [review]

**Pull request merged**